lib/src/base_impl.dart

```diff
@@ -9,83 +9,41 @@ import 'dart:convert';
 import 'exceptions.dart';
 import 'reddit.dart';
 
-String fullnameSync(RedditBase base) {
+String fullnameSync(RedditBaseInitializedMixin base) {
   if (base.data != null) {
     return base.data['name'];
   }
   return null;
 }
 
-void setData(RedditBase base, Map data) {
+void setData(RedditBaseInitializedMixin base, Map data) {
   base._data = data;
 }
 
-/// A base class for most DRAW objects which handles lazy-initialization of
-/// objects and Reddit API request state.
-abstract class RedditBase {
-  /// The current [Reddit] instance.
-  final Reddit reddit;
-
-  final RegExp _snakecaseRegExp = new RegExp("[A-Z]");
+abstract class RedditBaseInitializedMixin {
+  Reddit get reddit;
+  String get infoPath;
 
   /// Returns the raw properties dictionary for this object.
   ///
   /// This getter returns null if the object is lazily initialized.
   Map get data => _data;
-  Map _data;
-
-  /// The base request format for the current object.
-  String get infoPath => _infoPath;
-  String _infoPath;
+  Map _data = new Map();
 
   /// The fullname of a Reddit object.
   ///
   /// Reddit object fullnames take the form of 't3_15bfi0'.
-  Future<String> get fullname async => await property('name');
+  String get fullname => (data == null) ? null : data['name'];
```

Comment:
Could we make a asynchronous property method in the base_impl for all of our getters? It would make the code more readable, since to re-write the check for (data == null), I've been doing this check for Multireddit.dart as well, and it seems all our models will likely need to do this.